js/acolyte.js

```diff
@@ -17,57 +14,80 @@ self.onmessage = function onmessage(event) {
     switch (event.data.message) {
 
         case 'setup':
-            ({id, canvas, xOffset, xBounds, yBounds, maxDimensions, maxIters} = event.data)
-            ctx = canvas.getContext('2d')
-
+            ({ id, offset, bounds, dimensions, maxDimensions, maxIters } = event.data)
             draw()
             break
 
         case 'draw':
-            ({ xBounds, yBounds, maxIters } = event.data)
+            ({ bounds, maxIters } = event.data)
             draw()
     }
 }
 
 function draw() {
+    const start = performance.now();
+    const data = new Uint8ClampedArray(dimensions.width * dimensions.height * 4);
+    const colormap = makeColorMap(maxIters);
 
-    if (drawing) {
-        return
-    }
-
-    if (!canvas) {
-        return
-    }
-
-    drawing = true
+    const c = new Complex(0, 0);
 
-    for (let x = 0; x < canvas.width; x++) {
-        for (let y = 0; y < canvas.height; y++) {
+    for (let i = 0; i < data.length; i += 4) {
+        let x = offset.x + (i / 4) % dimensions.width;
+        let y = offset.y + (i / 4) / dimensions.width | 0;
```

Comment:
@lovasoa Forgot to ask earlier. What does that bitwise OR do?

Reply:
It does the same as `Math.floor`, but it's shorter :)

Reply:
@lovasoa Gotcha.